### DISCUSSION TITLE
Testing PHP Enum validation

### DISCUSSION THREAD

**Discussion by scottzirkel** [livewire/livewire#4915] [1. Help] [answered]
What is the best way to test PHP enum validation for Livewire? Here is an example of a custom rule:

``` php
protected function rules()
{
    return [
        'enum' => [new Enum(MyEnum::class)]
    ];
}
```

For the following enum:
``` php
enum MyEnum: string
{
    case Valid => 'valid';
}
```

The only thing I can get to pass right now is:

``` php
livewire(MyComponent::class)
    ->set('enum', 'invalid')
    ->assertHasErrors([
        'enum' => 'Illuminate\Validation\Rules\Enum',
    ]);
```
Above example in Pest.

That doesn't seem specific enough, is there a better option?

**Comment by joshhanley** [comment, marked as answer]
@scottzirkel looks ok to me. But do you know how you would test that in Laravel to be more specific? Livewire's method should be the same.

**Reply by scottzirkel** [reply]
To test it in Laravel I think I would do something similar but with `assertSessionHas`. I was just hoping for something that was more related to the particular enum I'm checking against vs just the rule itself. As long as this is best practice, I'm happy with it. I can always customize the return message. Thanks!

**Reply by joshhanley** [reply]
@scottzirkel yeah I'm not sure, haven't tested enums in Laravel yet either. I wonder if there is already a way to check the specific enum type and value? If there is then a Livewire test should be able to make use of that!
